File: 0027-remove-element/0027-remove-element.py
#VapsTech | 08/07/2024
#Complexity: O(n)
#Runtime: 33ms | Beats 80%
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums) #Get total length 

        count = 0
        for i in range(n):
            if nums[i] != val: #If it is not the val 
                nums[count] = nums[i] #Change nums without val
                count += 1 #Increment index

        return count #Return length of new arr without val

File: 0027-remove-element/0027-remove-element.cpp
//VapsTech | 08/07/2024
//Complexity: O(n)
//Runtime: 0ms | Beats 100%
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int count{0};

        for (int i = 0; i < n; i++){
            if (nums[i] != val){
                nums[count] = nums[i];
                count++;
            }
        }

        return count;
    }
};